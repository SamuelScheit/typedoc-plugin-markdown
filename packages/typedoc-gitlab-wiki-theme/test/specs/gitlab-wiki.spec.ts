import * as fs from 'fs';
import * as path from 'path';
describe(`GithubWiki`, () => {
  test(`should output index docs`, async () => {
    const contents = fs
      .readFileSync(path.join(__dirname, '../out/default/home.md'))
      .toString();
    expect(contents).toMatchSnapshot();
  });

  test(`should output member docs`, async () => {
    const contents = fs
      .readFileSync(
        path.join(__dirname, '../out/default/module-1/classes/ClassA.md'),
      )
      .toString();
    expect(contents).toMatchSnapshot();
  });

  test(`should output default sidebar`, async () => {
    const contents = fs
      .readFileSync(path.join(__dirname, '../out/default/_sidebar.md'))
      .toString();
    expect(contents).toMatchSnapshot();
  });

  test(`should output globals sidebar`, async () => {
    const contents = fs
      .readFileSync(path.join(__dirname, '../out/globals/_sidebar.md'))
      .toString();
    expect(contents).toMatchSnapshot();
  });
  test(`should output single modules sidebar`, async () => {
    const contents = fs
      .readFileSync(path.join(__dirname, '../out/single-modules/_sidebar.md'))
      .toString();
    expect(contents).toMatchSnapshot();
  });

  test(`should not output single page sidebar`, async () => {
    const contents = fs.readdirSync(path.join(__dirname, '../out/single-page'));
    expect(contents).toMatchSnapshot();
  });
});
