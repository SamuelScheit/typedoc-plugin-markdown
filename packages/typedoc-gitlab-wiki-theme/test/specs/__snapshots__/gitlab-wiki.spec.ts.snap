// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubWiki should not output single page sidebar 1`] = `
[
  "home.md",
]
`;

exports[`GithubWiki should output default sidebar 1`] = `
"## API

### module-1

#### Namespaces

- [Validation](module-1/namespaces/Validation/home)

#### Classes

- [ClassA](module-1/classes/ClassA)
- [ClassB](module-1/classes/ClassB)

#### Interfaces

- [InterfaceA](module-1/interfaces/InterfaceA)
- [InterfaceB](module-1/interfaces/InterfaceB)

### module-2

#### Classes

- [ClassA](module-2/classes/ClassA)
- [ClassB](module-2/classes/ClassB)

#### Interfaces

- [InterfaceA](module-2/interfaces/InterfaceA)
- [InterfaceB](module-2/interfaces/InterfaceB)
"
`;

exports[`GithubWiki should output globals sidebar 1`] = `
"## API

### Classes

- [ClassA](classes/ClassA)
- [ClassB](classes/ClassB)

### Interfaces

- [InterfaceA](interfaces/InterfaceA)
- [InterfaceB](interfaces/InterfaceB)
"
`;

exports[`GithubWiki should output index docs 1`] = `
"# typedoc-gitlab-wiki-theme

## Modules

- [module-1](./module-1/home)
- [module-2](./module-2/home)
"
`;

exports[`GithubWiki should output single modules sidebar 1`] = `
"## API

- [module-1](module-1/home)
- [module-2](module-2)
"
`;
